managing-os/templates/archlinux.md

```diff
@@ -76,22 +76,22 @@ Server = http://olivier.medoc.free.fr/archlinux/current/
 You should then follow the instruction related to pacman-key in order to sign the binary packages PGP key. With the key enabled, a pacman update will update qubes agents:
 ` # pacman -Suy `
 
-The two line that have just been added to /etc/pacman.conf should then be removed as they have been included in the qubes-vm-core update in the file `/etc/pacmand.d/99-qubes-repository-3.2.conf`
+The two lines that have just been added to /etc/pacman.conf should then be removed as they have been included in the qubes-vm-core update in the file `/etc/pacmand.d/99-qubes-repository-3.2.conf`
 
 ## Known Issues
 
 ### Package cannot be updated because of errors related to xorg-server or pulseaudio versions
 
-In case archlinux upgrade pulseaudio major version or xorg-server version, updating these packages will break the qubes GUI agent. To avoid breaking things, the update is blocked until a new version of the GUI agent is available.
+Upgrading pulseaudio or xorg-server to a newer major version will break the Qubes GUI agent. Avoid upgrading these packages until such time that a future version of the GUI agent addresses this issue.
```

Comment:
The way I read this, it appears the semantics have changed:

before: things break on major pulseaudio or *any* X upgrade
after: things break on major pulseaudio or *major* X upgrade

I agree that it sounds better to say this, however do you know that minor X upgrades are indeed fine? X not working is a rather unfortunate surprise.

Reply:
@jpouellet you're right, though I think the original wording is vague/possibly meant to be what I rephrased. We would have to query with the original author.

In an effort to correct the English semantics, can we afford to be a bit more vague just in case? Something like:

"There is a known issue affecting the Qubes GUI agent if pulseaudio or xorg-server are upgraded. To address this, avoid upgrading these packages until such time that a future version of the GUI agent addresses the issue."

It would be preferable to explain in more detail what the issue is, but without clarity on the major/minor versions, I think not mentioning either major/minor is better overall. What do you think?

Reply:
I understand the desire to avoid having precise but incorrect statements, but we must also be careful because generalizing statements as a rule would degrade the usefulness of the docs too.

If the underlying issue is something like we need a stable ABI to link against, and xorg guarantees that across minor versions but not major versions, then that is a useful and actionable detail to base decisions on.

Looks like this note was added by @TimFW in 099d9917 and edited by @ptitdoc in 1838cde6. Thoughts?

Reply:
Also ping @marmarek because you probably know the cause off the top of your head...

Reply:
X server: it define a lot of ABI versions, for different subsystems (video, input, etc), sometimes not changed on even major release
pulseaudio: modules are specific for X.Y version

In any case, Archlinux packaging define next major version as maximum supported (like `'xorg-server>=1.19.0' 'xorg-server<1.20.0'`, or `'pulseaudio>=11.0' 'pulseaudio<12.0'`), so minor upgrades are allowed, but major are not. Which actually may be wrong for pulseaudio...

Reply:
So should we instead say something like:

> "The Qubes GUI agent must be rebuilt whenever xorg-server or pulseaudio make major changes. If an update of one of these packages causes your template to break, simply [revert it](https://www.qubes-os.org/doc/software-update-vm/#reverting-changes-to-a-templatevm) and wait for corresponding Qubes package updates to be available (or attempt to build them yourself, if you're so inclined). This should not happen frequently."

?